src/freenas/etc/rc.conf.local

```diff
@@ -733,13 +733,33 @@ _gen_conf()
 		fi
 	done
 
-	echo "geli_devices=\"`${FREENAS_SQLITE_CMD} ${FREENAS_CONFIG} "SELECT encrypted_provider FROM storage_encrypteddisk e JOIN storage_volume v ON e.encrypted_volume_id = v.id WHERE v.vol_encrypt=1;" | \
-		tr \\\n \  `\""
-	${FREENAS_SQLITE_CMD} ${FREENAS_CONFIG} "SELECT e.encrypted_provider,v.vol_encryptkey FROM storage_encrypteddisk e JOIN storage_volume v ON e.encrypted_volume_id = v.id WHERE v.vol_encrypt=1;" | \
-	while read -r provider key; do
-		_provider=`echo ${provider}|tr '/-' '_'`
-		echo "geli_${_provider}_flags=\"-p -k /data/geli/${key}.key\""
-	done
+	# See redmine #31350
+	# Automatically mounting the geli providers at this stage in the boot process \
+	# causes problems for HA truenas.
+	# All TrueNAS HA customers are effected by this.
+	# 
+	# To get around this, we make a rudimentary check to make sure
+	# 1. there is no passphrase set
+	# 2. if there is no passphrase set, make sure it's not freenas or a single-node truenas
+	# 
+	# If any of the above conditions are met, then we mount the geli providers.
+	# Else we do not need to run this.
+
+        has_nopassphrase="$(${FREENAS_SQLITE_CMD} ${FREENAS_CONFIG} "SELECT vol_encrypt FROM storage_volume WHERE vol_encrypt = '1' ORDER BY -id LIMIT 1;")"
+        if [ "${has_nopassphrase}" = "1" ];
```

Comment:
@yocalebo  We are already filtering volumes without passphrase in the sql query below.